arrow-inject-annotations/src/main/kotlin/arrow/inject/annotations/Resolve.kt

```diff
@@ -8,4 +8,16 @@ package arrow.inject.annotations
 annotation class Resolve
 
 @CompileTime
-fun <A> context() = Unit
+fun <A> context(a: A? = null) = Unit
+
+@CompileTime
+fun <A, B> context(a: A? = null, b: B? = null) = Unit
+
+@CompileTime
+fun <A, B, C> context(a: A? = null, b: B? = null, c: C? = null) = Unit
+
+@CompileTime
+fun <A, B, C, D> context(a: A? = null, b: B? = null, c: C? = null, d: D? = null) = Unit
+
+@CompileTime
+fun <A, B, C, D, E> context(a: A? = null, b: B? = null, c: C? = null, d: D? = null, e: E? = null) = Unit
```

Comment:
Is it possible to avoid having to pass the value arguments to disambiguate it by suppressing the conflicting overload error in FIR and fixing the IR later?

Reply:
Or removing in Fir the value arguments so the user can't see they are there to be "passed"

Reply:
AFAIK I think this impl makes sense because suppressing diagnostic is not a public API in FIR at the moment.
Also de value arguments as laid out may work for overrides:
```kotlin
context <A, B>(b = overridenB)
```